test/framework/modules.py

```diff
@@ -404,6 +404,10 @@ def test_modpath_extensions_for(self):
         error_pattern = "Can't get value from a non-existing module"
         self.assertErrorRegex(EasyBuildError, error_pattern, self.modtool.modpath_extensions_for, ['nosuchmodule/1.2'])
 
+        realhome = os.environ.get('HOME')
+        os.environ['HOME'] = os.path.join(self.test_prefix, 'HOME')
+        os.makedirs("%s/modules/Compiler/GCC/4.7.2" % os.environ['HOME'])
```

Comment:
OK, but what if the directory is created later? You still want to exclude the module that may start extending `$MODULEPATH` with `.../Compiler/GCC/4.7.2` in the future when the corresponding directory is created, no?